### PR TITLE
Issue 5446 - Fix some covscan issues

### LINKED ISSUE #5446: Fix some covscan issues
**Issue Description**
Civscan reports a number of "Important" issues:
      2	BAD_FREE                                        	incorrect_free
      3	BUFFER_SIZE                                     	buffer_size_warning
      6	COPY_PASTE_ERROR                                	copy_paste_error
      3	CPPCHECK_WARNING                                	error[resourceLeak]
      2	IDENTICAL_BRANCHES                              	identical_branches
     13	RESOURCE_LEAK                                   	leaked_storage
      1	RESOURCE_LEAK                                   	overwrite_var
      4	USE_AFTER_FREE                                  	deref_arg
      3	USE_AFTER_FREE                                  	double_free
      1	USE_AFTER_FREE                                  	pass_freed_arg
Details are in  [https://cov01.lab.eng.brq2.redhat.com/covscanhub/task/260426/log/389-ds-base-2.3.0-202209121502gitd5ba53748.fc36/scan-results-imp.html](https://cov01.lab.eng.brq2.redhat.com/covscanhub/task/260426/log/389-ds-base-2.3.0-202209121502gitd5ba53748.fc36/scan-results-imp.html)

**Package Version and Platform:**
 - Platform: Fedora
 - Package and version: main branch on 2022, 12 Sept

**Steps to Reproduce**

1. SKIP_AUDIT_CI=1 make -f rpm.mk srpms
2. covscan mock-build --config=fedora-36-x86_64 ./dist/srpms/389-ds-base-*.src.rpm

### PR DESCRIPTION
**Pull request by progier389**
Problem: 
   covscan reported a number of important issues.

Solution:
Most of those were false positive but a few were genuine errors. I annotated the false positive so the issues does not get reported ( except the COPY_PASTE_ERROR whose annotation seems ignored (coverity bug?) but at least the comment visible in the detailed report shows it is a false positive without even having to look at the code)

Now the scan-results-imp-summary.txt is getting minimal:
Except copy_[paste_errors only 2 errors in jemalloc third party code are reported.

Issue: [5446](https://github.com/389ds/389-ds-base/issues/5446)

Reviewed by: @mreynolds389 , @droideck

### CODE REVIEW

File: lib/libaccess/usrcache.cpp [hunk 1]
```diff
@@ -174,14 +174,18 @@ int acl_usr_cache_init ()
...
+	    usrobj = (UserCacheObj *)pool_malloc(usrcache_pool,
 					     sizeof(UserCacheObj));
 
-	if (!usrobj) return -1;
-	memset((void *)usrobj, 0, sizeof(UserCacheObj));
-	PR_INSERT_AFTER(&usrobj->list, usrobj_list);
+	    if (!usrobj) return -1;
+	    memset((void *)usrobj, 0, sizeof(UserCacheObj));
+	    PR_INSERT_AFTER(&usrobj->list, usrobj_list);
     }
```
Comment (mreynolds389) [inline lib/libaccess/usrcache.cpp:185]: indentation is off in this function

**Review by mreynolds389 — APPROVED** [review]
minor indention issue, but the rest looks good!

File: lib/libaccess/usrcache.cpp (hunk 1 above)
Comment (progier389) [reply lib/libaccess/usrcache.cpp:185]: Fixed that one and also removed the Cargo file ...

File: lib/libaccess/usrcache.cpp [hunk 2]
```diff
@@ -174,14 +174,18 @@ int acl_usr_cache_init ()
...
-	usrobj = (UserCacheObj *)pool_malloc(usrcache_pool,
-					     sizeof(UserCacheObj));
-
-	if (!usrobj) return -1;
-	memset((void *)usrobj, 0, sizeof(UserCacheObj));
-	PR_INSERT_AFTER(&usrobj->list, usrobj_list);
+        /* covscan false positive usrobj is linked in the cache */
+        /* coverity[overwrite_var] */
+        usrobj = (UserCacheObj *)pool_malloc(usrcache_pool,
+         sizeof(UserCacheObj));
```
Comment (droideck) [inline lib/libaccess/usrcache.cpp]: Minor indentation issue. I think we usually try to keep the indentation like this:

```
        usrobj = (UserCacheObj *)pool_malloc(usrcache_pool,
                                             sizeof(UserCacheObj));
```

**Review by droideck — APPROVED** [review]
Very minor issue :)
The rest LGTM!

**Pull request merged**